packages/admin/admin-theme/src/componentsTheme/MuiDataGrid.tsx

```diff
@@ -13,12 +13,36 @@ import {
     TextField,
     TextFieldProps,
 } from "@mui/material";
+import { Spacing } from "@mui/system";
 import { getDataGridUtilityClass, GRID_DEFAULT_LOCALE_TEXT, gridClasses } from "@mui/x-data-grid";
 import type {} from "@mui/x-data-grid/themeAugmentation";
 
 import { mergeOverrideStyles } from "../utils/mergeOverrideStyles";
 import { GetMuiComponentTheme } from "./getComponentsTheme";
 
+const getDensityStyles = (density: string | unknown, spacing: Spacing) => {
+    switch (density) {
+        case "compact":
+            return {
+                minHeight: `${spacing(8)} !important`,
+                maxHeight: `${spacing(8)} !important`,
+                height: `${spacing(8)} !important`,
+            };
+        case "comfortable":
+            return {
+                minHeight: `${spacing(16)} !important`,
+                maxHeight: `${spacing(16)} !important`,
+                height: `${spacing(16)} !important`,
+            };
+        default:
+            return {
+                minHeight: `${spacing(12)} !important`,
+                maxHeight: `${spacing(12)} !important`,
+                height: `${spacing(12)} !important`,
+            };
+    }
+};
```

Comment:
I don't think each of those values needs to be set in every case, maybe not even always with `!important`. 
You could have this function only return the height-value then set it in the individual selectors, as needed.